Source/ROSIntegrationVision/Private/VisionComponent.cpp

```diff
@@ -0,0 +1,755 @@
+	// * - Depth image data (width * height * 2 Bytes (Float16))
+	uint8_t* DepthPtr = &Priv->Buffer->Read[OffsetDepth];
+	uint32_t TargetDepthBufSize = Width*Height * 4;
+	uint8_t* TargetDepthBuf = new uint8_t[TargetDepthBufSize]; // Allocate a byte for every pixel * 4 Bytes for a single 32Bit Float
+
+	const uint32_t ColorImageSize = Width * Height * 3;
+	convertDepth((uint16_t *)DepthPtr, (__m128*)TargetDepthBuf);
+	// convertDepth((uint16_t *)packet.pDepth, (__m128*)&msgDepth->data[0]);
+
+	UE_LOG(LogTemp, Display, TEXT("Buffer Offsets: %d %d %d"), OffsetColor, OffsetDepth, OffsetObject);
+
+	FROSTime time = FROSTime::Now();
+
+	TSharedPtr<ROSMessages::sensor_msgs::Image> ImageMessage(new ROSMessages::sensor_msgs::Image());
+
+	ImageMessage->header.seq = 0;
+	ImageMessage->header.time = time;
+	ImageMessage->header.frame_id = TEXT("/unreal_ros/image_optical_frame");
+	ImageMessage->height = Height;
+	ImageMessage->width = Width;
+	ImageMessage->encoding = TEXT("bgr8");
+	ImageMessage->step = Width * 3;
+	ImageMessage->data = &Priv->Buffer->Read[OffsetColor];
+	_ImagePublisher->Publish(ImageMessage);
+
+	TSharedPtr<ROSMessages::sensor_msgs::Image> DepthMessage(new ROSMessages::sensor_msgs::Image());
+
+	DepthMessage->header.seq = 0;
+	DepthMessage->header.time = time;
+	DepthMessage->header.frame_id = TEXT("/unreal_ros/image_optical_frame");
+	DepthMessage->height = Height;
+	DepthMessage->width = Width;
+	DepthMessage->encoding = TEXT("32FC1");
+	DepthMessage->step = Width * 4;
+	DepthMessage->data = TargetDepthBuf;
+	_DepthPublisher->Publish(DepthMessage);
+
+	Priv->Buffer->DoneReading();
+
+	double x = Priv->Buffer->HeaderRead->Translation.X;
+	double y = Priv->Buffer->HeaderRead->Translation.Y;
+	double z = Priv->Buffer->HeaderRead->Translation.Z;
+	double rx = Priv->Buffer->HeaderRead->Rotation.X;
+	double ry = Priv->Buffer->HeaderRead->Rotation.Y;
+	double rz = Priv->Buffer->HeaderRead->Rotation.Z;
+	double rw = Priv->Buffer->HeaderRead->Rotation.W;
+
+	if (!FixedCam) {
+		TSharedPtr<ROSMessages::tf2_msgs::TFMessage> TFMessage(new ROSMessages::tf2_msgs::TFMessage());
+		ROSMessages::geometry_msgs::TransformStamped TransformStamped;
+		TransformStamped.header.seq = 0;
+		TransformStamped.header.time = time;
+		TransformStamped.header.frame_id = ParentLink;
+		TransformStamped.child_frame_id = TEXT("/unreal_ros/image_frame");
+		TransformStamped.transform.translation.x = x;
+		TransformStamped.transform.translation.y = y;
+		TransformStamped.transform.translation.z = z;
+		TransformStamped.transform.rotation.x = rx;
+		TransformStamped.transform.rotation.y = ry;
+		TransformStamped.transform.rotation.z = rz;
+		TransformStamped.transform.rotation.w = rw;
+
+		TFMessage->transforms.Add(TransformStamped);
+
+		_TFPublisher->Publish(TFMessage);
+	}
+	// Publish optical frame
+	FRotator CameraLinkRotator(0.0, -90.0, 90.0);
+	FQuat CameraLinkQuaternion(CameraLinkRotator);
+
+	if (!FixedCam) {
```

Comment:
This if block can also go into the last if(!(FixedCam) block. I think there is no need to have to if-blocks